### PR TITLE
server: support nodes API endpoint for secondary shared tenants

### LINKED ISSUE #110023: server: the "nodes" status API endpoint should work with secondary tenants with capability `can_view_node_info`
Found while investigating #110001.

**Describe the problem**

When `TestNodeStatusResponse` in `server/storage_api` is modified to point to a secondary tenant (e.g. via #110008), the test starts failing when accessing the `/_status/nodes` endpoint:

```
    nodes_test.go:113: expected privilege error, got status: 501 Not Implemented, content-type: application/json, body: {
          "error": "method Nodes not implemented",
          "code": 12,
          "message": "method Nodes not implemented",
          "details": [
          ]
        }, error: <nil>
```

**How to reproduce**

from #110008 reset `DefaultTestTenant` in `TestNodeStatusResponse`, run the test with a test tenant and observe the failure.

**Expected behavior**

Once the tenant is granted a suitable capability, the endpoint should work properly. It should retrieve the node details over the KV tenant connector over the network.

Epic CRDB-38968

Jira issue: CRDB-31228

### PR DESCRIPTION
**Pull request by cthumuluru-crdb**
The "nodes" status API endpoint should work with "shared" secondary tenants. Since nodes are common to all the tenants, this API endpoint behaves similar to that of system tenant. Shared service tenants implicitly have all capabilities.

Tenant connector implements TenantStatusServer interface. Status server in secondary tenants make use of tenant connector to fetch nodes information.

Support for external tenants is tracked in a separate ticket.

Fixes: https://github.com/cockroachdb/cockroach/issues/110023
Epic: [CRDB-38968](https://cockroachlabs.atlassian.net/browse/CRDB-38968)
Release note (multi-tenancy): The "nodes" endpoint should work for "shared" secondary tenants. Since nodes are common to all the tenants, this API endpoint behaves similar to that of system tenant.

### CODE REVIEW

**Review by dhartunian — APPROVED** [review]

File: pkg/server/status.go [hunk 1]
```diff
@@ -1539,6 +1539,25 @@ func (s *statusServer) Stacks(
 	return stacksLocal(req)
 }
 
+// Nodes returns all node statuses for a secondary tenant with
+// `can_view_node_info` capability.
```
Comment (stevendanna) [inline pkg/server/status.go:1543]: Do we have a test that covers that this endpoint fails without the capability? If not, it may be worth it to add one.

**Review by stevendanna — APPROVED** [review]

File: pkg/server/status.go (hunk 1 above)
Comment (cthumuluru-crdb) [reply pkg/server/status.go:1543]: I was using the test mentioned in the ticket that was failing before my changes. I tried fixing the test to work for negative case too but found out `can_view_node_info` has no affect on shared tenants. I tested the changes for "external" secondary tenants and it was works as expected. 

On digging further, I found https://github.com/cockroachdb/cockroach/blob/75a1f36daeebcaa1bf9333f1912b83c2b7a54472/pkg/multitenant/tenantcapabilities/tenantcapabilitiesauthorizer/authorizer.go#L469-L475 

From the above comment it looks like shared tenants implicitly have all capabilities unless we introduce a new mode which I believe will be a bigger change.

Comment (stevendanna) [reply pkg/server/status.go:1538]: Yup, sorry for not being a bit more explicit. I think we can probably write a test that explicitly starts a separate process tenant and then makes this API call.  You can do this either by passing `ExternalTestTenantAlwaysEnabled` as the value of `DefaultTestTenant` during TestServer startup, or by passing `TestControlsTenantsExplicitly` and then using the methods on TestServer to start a separate process test server manually.  Putting this test in place in the server package will probably be useful as we can add more of these negative cases to it as we enable more endpoints.

Comment (cthumuluru-crdb) [reply pkg/server/status.go:1538]: https://github.com/cockroachdb/cockroach/blob/b617593ae2d5013035b3633f95c585948471d2d3/pkg/server/storage_api/nodes_test.go#L93-L103
I created the negative test for external tenant using a table. Let me know if you prefer a separate test for the external mode.

File: pkg/server/storage_api/nodes_test.go [hunk 2]
```diff
@@ -83,62 +83,78 @@ func TestStatusJson(t *testing.T) {
...
-	s := srv.ApplicationLayer()
+	testCases := []testCase{
+		{
+			name:                     "test_node_status_reponse_shared_tenant",
+			defaultTestTenantOptions: base.SharedTestTenantAlwaysEnabled,
+		},
+		{
+			name:                     "test_node_status_reponse_external_tenant",
+			defaultTestTenantOptions: base.ExternalTestTenantAlwaysEnabled,
+			expectedErr:              "status: 500", // `can_view_node_info` capability is missing
```
Comment (stevendanna) [inline pkg/server/storage_api/nodes_test.go:101]: Thanks for adding this test. 500 for a permissions error is sad, but that is an existing problem that I think we have an issue for somewhere.

**Review by stevendanna — APPROVED** [review]

File: pkg/cli/testdata/zip/testzip_external_process_virtualization [hunk 3]
```diff
@@ -85,9 +85,7 @@ debug zip --concurrency=1 --cpu-profile-duration=1s /dev/null
 [cluster] retrieving SQL data for system.tenant_tasks... writing output: debug/system.tenant_tasks.txt... done
 [cluster] retrieving SQL data for system.tenant_usage... writing output: debug/system.tenant_usage.txt... done
 [cluster] retrieving SQL data for system.tenants... writing output: debug/system.tenants.txt... done
-[cluster] requesting nodes... received response...
-[cluster] requesting nodes: last request failed: rpc error: ...
-[cluster] requesting nodes: creating error output: debug/nodes.json.err.txt... done
+[cluster] requesting nodes... received response... writing JSON output: debug/nodes.json... done
```
Comment (cthumuluru-crdb) [inline pkg/cli/testdata/zip/testzip_external_process_virtualization:88]: @stevendanna - sorry for the noise on this PR. After pushing the changes I notices a couple of test failures in the CI run. They are related to data driven tests. I fixed the expected zip log to be inline with the code fixes done. 

More importantly, I was curious why `testzip_external_process_virtualization` test was not failing with capability missing error. After carefully looking at both the test for differences, data driven tests in `zip_tenant_test.go` are running in insecure mode due to https://github.com/cockroachdb/cockroach/issues/77173 and that's the reason they were passing without any issues. 
https://github.com/cockroachdb/cockroach/blob/c68c559859be738efead9971f5e11f62a8c69d06/pkg/cli/zip_tenant_test.go#L76-L86 
I'll add this issue to CRDB-38970 it is not already being tracked. This PR is ready to be committed now. Can you please take a look and bless it one more time?

**Review by stevendanna — APPROVED** [review]

**Pull request merged**